hazelcast-client-new/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -170,39 +185,55 @@ private void parseAndBuildConfig(ClientConfig clientConfig) throws Exception {
             domLevel3 = false;
         }
         process(root);
+        schemaValidation(root.getOwnerDocument());
         handleConfig(root);
     }
 
     private void handleConfig(final Element docElement) throws Exception {
         for (Node node : new IterableNodeList(docElement.getChildNodes())) {
             final String nodeName = cleanNodeName(node.getNodeName());
-            if ("security".equals(nodeName)) {
-                handleSecurity(node);
-            } else if ("proxy-factories".equals(nodeName)) {
-                handleProxyFactories(node);
-            } else if ("properties".equals(nodeName)) {
-                fillProperties(node, clientConfig.getProperties());
-            } else if ("serialization".equals(nodeName)) {
-                handleSerialization(node);
-            } else if ("native-memory".equals(nodeName)) {
-                fillNativeMemoryConfig(node, clientConfig.getNativeMemoryConfig());
-            } else if ("group".equals(nodeName)) {
-                handleGroup(node);
-            } else if ("listeners".equals(nodeName)) {
-                handleListeners(node);
-            } else if ("network".equals(nodeName)) {
-                handleNetwork(node);
-            } else if ("load-balancer".equals(nodeName)) {
-                handleLoadBalancer(node);
-            } else if ("near-cache".equals(nodeName)) {
-                handleNearCache(node);
-            } else if ("executor-pool-size".equals(nodeName)) {
-                final int poolSize = Integer.parseInt(getTextContent(node));
-                clientConfig.setExecutorPoolSize(poolSize);
+            if (occurrenceSet.contains(nodeName)) {
+                throw new InvalidConfigurationException("Duplicate '" + nodeName + "' definition found in XML configuration. ");
+            }
+            handleXmlNode(node, nodeName);
+            if (!canOccurMultipleTimes(nodeName)) {
+                occurrenceSet.add(nodeName);
             }
         }
     }
 
+    private void handleXmlNode(Node node, String nodeName) throws Exception {
+
+        if (SECURITY.isEqual(nodeName)) {
+            handleSecurity(node);
+        } else if (PROXY_FACTORIES.isEqual(nodeName)) {
+            handleProxyFactories(node);
+        } else if (PROPERTIES.isEqual(nodeName)) {
+            fillProperties(node, clientConfig.getProperties());
+        } else if (SERIALIZATION.isEqual(nodeName)) {
+            handleSerialization(node);
+        } else if (NATIVE_MEMORY.isEqual(nodeName)) {
+            fillNativeMemoryConfig(node, clientConfig.getNativeMemoryConfig());
```

Comment:
I would give all these methods the same name. I would go for parse.
so parseHandleSecurity, parseGroup, parseExecutorPoolSize etc.

If you make the method names the same, then there is no confusion for other people reading/maintaining your code.